stan/math/prim/fun/as_array_or_scalar.hpp

```diff
@@ -1,41 +1,38 @@
-#ifndef STAN_MATH_PRIM_META_AS_ARRAY_OR_SCALAR_HPP
-#define STAN_MATH_PRIM_META_AS_ARRAY_OR_SCALAR_HPP
+#ifndef STAN_MATH_PRIM_FUN_AS_ARRAY_OR_SCALAR_HPP
+#define STAN_MATH_PRIM_FUN_AS_ARRAY_OR_SCALAR_HPP
 
 #include <stan/math/prim/fun/Eigen.hpp>
-#include <stan/math/prim/meta/holder.hpp>
-#include <stan/math/prim/meta/is_vector.hpp>
-#include <stan/math/prim/meta/is_eigen.hpp>
-#include <stan/math/prim/meta/is_stan_scalar.hpp>
+#include <stan/math/prim/meta.hpp>
 #include <vector>
 
 namespace stan {
 namespace math {
 
-/** \ingroup type_trait
+/**
  * Returns specified input value.
  *
  * @tparam T Type of element.
  * @param v Specified value.
  * @return Same value.
  */
-template <typename T, typename = require_stan_scalar_t<T>>
+template <typename T, require_stan_scalar_t<T>* = nullptr>
 inline T as_array_or_scalar(T&& v) {
   return std::forward<T>(v);
 }
 
-/** \ingroup type_trait
+/**
  * Converts a matrix type to an array.
  *
  * @tparam T Type of \c Eigen \c Matrix or expression
  * @param v Specified \c Eigen \c Matrix or expression.
  * @return Matrix converted to an array.
  */
-template <typename T, typename = require_eigen_t<T>>
+template <typename T, require_eigen_t<T>* = nullptr>
 inline auto as_array_or_scalar(T&& v) {
   return make_holder([](auto& x) { return x.array(); }, std::forward<T>(v));
```

Comment:
It is often used like 
```auto x = as_array_or_scalar(something);```
If that something is a rvalue (for example a function call) this would not work without holder. I am starting to think adding a require to simply reject rvalues might be better. Not sure tho.